lib/ITip/Broker.php

```diff
@@ -286,16 +286,43 @@ protected function processMessageRequest(Message $itipMessage, VCalendar $existi
                 $existingObject->add(clone $component);
             }
         } else {
-            // We need to update an existing object with all the new
-            // information. We can just remove all existing components
-            // and create new ones.
+            // We need to update an existing object with all the new information.
+            // We start by removing all non-VEVENT components (VTIMEZONE, etc.)
             foreach ($existingObject->getComponents() as $component) {
-                $existingObject->remove($component);
+                if ($component->name !== 'VEVENT') {
+                    $existingObject->remove($component);
```

Comment:
Yeah, I wondered too while reading the code. 
Previously `VTIMEZONE` components were also removed, and added back from the iTip. 

I'm fine reworking this to only remove and add `VEVENT` components. Let me know